src/react/hooks.ts

```diff
@@ -98,6 +98,30 @@ export function useElementData(id: string): WorkbookElementData {
   return data;
 }
 
+/**
+ * Provides the latest data values from corresponding sheet with a callback to fetch more
+ * @param {string} id Sheet ID to get element data from
+ * @returns {WorkbookElementData} Element Data for corresponding sheet, if any
+ */
+export function usePaginatedElementData(id: string): WorkbookElementData {
+  const client = usePlugin();
+  const [data, setData] = useState<WorkbookElementData>({});
+
+  const loadMore = useCallback(() => {
+    if (id) {
+      client.elements.fetchMoreElementData(id);
+    }
+  }, [id]);
+
+  useEffect(() => {
+    if (id) {
+      return client.elements.subscribeToElementData(id, setData);
+    }
+  }, [client, id]);
+
+  return useMemo(() => [data, loadMore], [data]);
```

Comment:
The expectation is that the user should immediately destructure the result of this hook, so it doesn't need to be memo-ed

```suggestion
  return [data, loadMore];
```